### PR TITLE
Contacts polishing

### PR DESCRIPTION
**Pull request by SamTremko**

### CODE REVIEW

**Comment by SamTremko** [comment]
* **#1113** <a href="https://app.graphite.dev/github/pr/vexl-it/vexl/1113?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @SamTremko and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/mobile/src/components/RootNavigation/index.tsx [hunk 1]
```diff
@@ -92,7 +92,11 @@ function RootNavigation(): JSX.Element {
             <Stack.Screen name="FilterOffers" component={FilterOffersScreen} />
             <Stack.Screen name="AppLogs" component={AppLogsScreen} />
             <Stack.Screen name="ChatDetail" component={ChatDetailScreen} />
-            <Stack.Screen name="SetContacts" component={SetContactsScreen} />
+            <Stack.Screen
+              name="SetContacts"
+              component={SetContactsScreen}
+              initialParams={{}}
```
Comment (kaladivo) [inline apps/mobile/src/components/RootNavigation/index.tsx]: Remove

File: apps/mobile/src/components/SetContactsScreen/components/ContactListSelect/atom.ts [hunk 2]
```diff
@@ -26,101 +28,165 @@ import userSvg from '../../../images/userSvg'
...
+          ? selectedSubmittedNumbersAtom
+          : contactsFilter === 'nonSubmitted'
+          ? selectedNonSubmittedNumbersAtom
+          : selectedNewNumbersAtom
+      )
+    },
+    (
+      get,
+      set,
+      numbers: (value: Set<E164PhoneNumber>) => Set<E164PhoneNumber>
```
Comment (kaladivo) [inline apps/mobile/src/components/SetContactsScreen/components/ContactListSelect/atom.ts]: Remove callback and call atom at line 164 if necessary

File: apps/mobile/src/components/SetContactsScreen/components/ContactListSelect/components/SearchBar.tsx [hunk 3]
```diff
@@ -1,47 +1,45 @@
...
+//         Your word is{' '}
+//         <Text color="$black" fontWeight="800">
+//           empower
+//         </Text>
+//         💪.
+//       </Text>
+//       <Text color="$black">Good luck finding the rest! 🚀</Text>
+//     </Stack>
+//   )
+// }
```
Comment (kaladivo) [inline apps/mobile/src/components/SetContactsScreen/components/ContactListSelect/components/SearchBar.tsx]: Remove completely

File: apps/mobile/src/components/SetContactsScreen/components/ContactListSelect/components/SearchBar.tsx [hunk 4]
```diff
@@ -63,30 +61,30 @@ function SearchBar(): JSX.Element {
...
+  //           },
+  //         ],
+  //       }),
+  //       TE.map(() => {
+  //         setSearchText('')
+  //         setInputValue('')
+  //       })
+  //     )()
+  //   }
+  // }, [searchText, setSearchText, showModal])
```
Comment (kaladivo) [inline apps/mobile/src/components/SetContactsScreen/components/ContactListSelect/components/SearchBar.tsx]: Also remove

File: apps/mobile/src/components/SetContactsScreen/components/ContactListSelect/index.tsx [hunk 5]
```diff
@@ -48,74 +71,87 @@ function ContactsListSelect({
...
-        initialFilters: {
-          showNew: props.showNewByDefault,
-          showNonSubmitted: false,
-          showSubmitted: false,
-        },
-        reloadContacts: () => {
-          setReloadContacts((v) => v + 1)
-        },
-      }}
+      value={useMemo(
```
Comment (kaladivo) [inline apps/mobile/src/components/SetContactsScreen/components/ContactListSelect/index.tsx]: use callback if it needs to be memoized

File: apps/mobile/src/state/lastRouteMmkvAtom.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import {z} from 'zod'
+import {atomWithParsedMmkvStorage} from '../utils/atomUtils/atomWithParsedMmkvStorage'
+
+const wasLastRouteBeforeRedirectOnContactsScreenMmkvAtom =
```
Comment (kaladivo) [inline apps/mobile/src/state/lastRouteMmkvAtom.ts:4]: remove on logout

**Review by kaladivo — APPROVED** [review]

**Pull request merged**